cluster/common.sh

```diff
@@ -202,6 +202,16 @@ function gen-kube-bearertoken() {
     KUBE_BEARER_TOKEN=$(dd if=/dev/urandom bs=128 count=1 2>/dev/null | base64 | tr -d "=+/" | dd bs=32 count=1 2>/dev/null)
 }
 
+# Generate uid
+# This function only works on systems with python. It generates a time based
+# UID instead of a UUID because GCE has a name length limit.
+#
+# Vars set:
+#   KUBE_UID
+function gen-uid {
+    KUBE_UID=$(python -c 'import uuid; print uuid.uuid1().fields[0]')
```

Comment:
Maybe consider this? Your choice. I am okay with python. 
`cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1`